src/GovukNotify/Client/BaseClient.cs

```diff
@@ -98,15 +115,15 @@ private async Task<HttpResponseMessage> SendRequest(string url, HttpMethod metho
 
             try
             {
-                response = await this.client.SendAsync(request).ConfigureAwait(false);
+                response = await this.client.SendAsync(request, cancellationToken).ConfigureAwait(false);
             }
             catch (AggregateException ae)
             {
                 ae.Handle(x =>
                 {
                     if (x is HttpRequestException)
                     {
-                        throw new NotifyClientException(x.InnerException.Message);
+                        throw new NotifyClientException(x.InnerException?.Message ?? x.Message);
```

Comment:
It's possible for `InnerException` to be null here, which would lead to an uncaught `NullReferenceException`.